lib/validators/upload_validator.rb

```diff
@@ -44,6 +46,16 @@ def validate(upload)
     end
   end
 
+  # this should only be run on existing records, and covers cases of
+  # upload.update_secure_status being run outside of the creation flow,
+  # where some cases e.g. have exemptions on the extension enforcement
+  def changing_upload_security?(upload)
+    !upload.new_record? && \
+      upload.secure_changed? && \
+      upload.security_last_changed_at_changed? && \
+      upload.security_last_changed_reason_changed?
```

Comment:
I think it would be safer to have explicit list.

If I change for existing record secure, security_last_changed_at, security_last_changed_reason AND url validation will pass. 

Also, when one of the values will stay same, `_changed?` will return false. 

Maybe it would be better to have allow list, something like:
```
upload.changed_attributes.keys.all {|attribute| %i(secure, security_last_changed_at, security_last_changed_reason).includes?(attribute)}
```

Reply:
Ah yeah I see what you are saying, if `upload.changed_attributes` are all those 3. This combination of 3 changing is special because `update_secure_status` updates all 3 at once. Will change!